feature-libs/product-multi-dimensional/selector/core/services/product-multi-dimensional-selector-images.service.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { TestBed } from '@angular/core/testing';
+import { Config, VariantOptionQualifier } from '@spartacus/core';
+import { ProductMultiDimensionalSelectorImagesService } from './product-multi-dimensional-selector-images.service';
+import { ProductMultiDimensionalConfig } from '../../root/config/product-multi-dimensional-config';
+
+enum VariantQualifier {
+  STYLE_SWATCH = 'styleSwatch',
+}
+
+describe('ProductMultiDimensionalSelectorImagesService', () => {
+  let service: ProductMultiDimensionalSelectorImagesService;
+  let config: Config;
+
+  beforeEach(() => {
+    const mockConfig = {
+      backend: {
+        media: {
+          baseUrl: 'http://media.base.url/',
+        },
+        occ: {
+          baseUrl: 'http://occ.base.url/',
+        },
+      },
+      multiDimensional: {
+        imageFormat: VariantQualifier.STYLE_SWATCH,
+      } as ProductMultiDimensionalConfig,
+    };
+
+    TestBed.configureTestingModule({
+      providers: [
+        ProductMultiDimensionalSelectorImagesService,
+        { provide: Config, useValue: mockConfig },
+      ],
+    });
+
+    service = TestBed.inject(ProductMultiDimensionalSelectorImagesService);
+    config = TestBed.inject(Config);
+  });
+
+  describe('getVariantOptionImages', () => {
+    it('should return images that match the specified format', () => {
+      config.multiDimensional.imageFormat = VariantQualifier.STYLE_SWATCH;
```

Comment:
```suggestion
      config.multiDimensional = { imageFormat: VariantQualifier.STYLE_SWATCH };
```
To avoid error `'config.multiDimensional' is possibly 'undefined'.`

Reply:
fixed